lib/definition/value_object.rb

```diff
@@ -19,6 +19,10 @@ def initialize(args = nil, **kwargs)
       super(result.value.freeze)
     end
 
+    def new(args)
+      self.class.new(merge(args))
```

Comment:
this will only work if the value object is a hash, it can however also be an array for example (see https://github.com/Goltergaul/definition/blob/06a388d17911ac77a666056314ce0fab814e9696/spec/lib/definition/value_object_spec.rb#L18)

However I agree that that functionality is indeed missing. I was thinking already to change the value object a bit and maybe make it hash only. That would simplify things a lot

Reply:
I'd suggest we stick our heads together quickly when you are back from vacations

Reply:
Somehow i thought so, but the tests were passing ;-). Yeah let's have a look after my vacation.